src/uwtools/api/chgres_cube.py

```diff
@@ -32,9 +34,12 @@ def execute(
     :param batch: Submit run to the batch system?
     :param dry_run: Do not run the executable, just report what would have been done.
     :param graph_file: Write Graphviz DOT output here.
+    :param stdin_ok: OK to read from stdin?
     :return: ``True`` if task completes without raising an exception.
     """
-    obj = _ChgresCube(config=config, cycle=cycle, batch=batch, dry_run=dry_run)
+    obj = _ChgresCube(
+        config=_ensure_data_source(config, stdin_ok), cycle=cycle, batch=batch, dry_run=dry_run
```

Comment:
The `config` value will be a `Path` if its was initially a `str`, or otherwise whatever value was originally supplied (a `dict`, `Config`, or `Path`) -- unless it was `None` and `stdin_ok` was not set, in which case an exception will be raised.

A lot of the same follows... I'll comment on interesting changes.